### ISSUE TITLE
Is isDefault correctly false?

### ISSUE THREAD

**Issue by kalosisz** [nielsbasjes/yauaa#312]
**The exact useragent**
useragent: 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:62.0) Gecko/20100101 Firefox/62.0'

**Expected/Incorrect fields**
OperatingSystemVersion agentField
Obtained { value:'??', confidence:'149', default:'??', isDefault:false }
Expected { value:'??', confidence:'149', default:'??', isDefault:true }

**Expected behavior**
I'd expect to have the isDeafult field set to true if the user agent field value is equal to the deafult of the field. 

Is the current implementation correct if the isDeafult is set to false?

**Comment by nielsbasjes** [comment]
Good question;
In this system you can have a "default value" (indicating the property is to be treated as `we don't know` / unknown) and a "normal value".

Also in setting a value I can set something to the special value `<<<null>>>` which simply means that we have a certainty about the fact we do not know what it is. In case we are more certain about not knowing than about a value that was found.

I quite sure you have found a place where I set a value (`??`) instead of saying `I don't know` which (depending on the field) also becomes something like `Unknown` or `??`.

I'll double check this example and possibly other similar cases.

**Comment by nielsbasjes** [comment]
So I went through the code to remember why I did this. Let me try to explain.
I have several possible outcomes when a Matcher matches and specifies something for a field:

1. We have a value with a confidence. I.e. it is `Mozilla` with confidence 42.
2. We `Don't know` also with a confidence.  
     a. Most of the times we allow  calculator to provide a fallback (like a brand derived from a url domain), so we use the `<<<null>>>` value which `isDefault`.
     b. Sometimes we really want to force `Unknown` because we do not want these fallbacks.  

And ... in some cases I used 2b (which is not `isDefault`) where I probably should have used 2a instead.
These do not surface because there is no difference if there are no postprocessing calculations that impact it.

I'm going to put this in the documentation and fix the rules.
